src/get_imports/mod.rs

```diff
@@ -73,78 +70,61 @@ fn get_imports_from_file<'a>(
-                        .filter_map(|spec| -> Option<String> {
-                            return get_specifier_name(&fm, spec);
-                        })
-                        .collect();
-                    let import_source = i.src.value.to_string();
-                    match imports_map.get(&import_source) {
-                        Some(Some(current_set)) => {
-                            // TODO find a way to append data to current set instead of copying it into an new one
-                            let mut new_set: HashSet<String> =
-                                HashSet::from_iter(current_set.iter().map(|v| v).cloned());
-                            for val in specs_name {
-                                new_set.insert(val);
-                            }
-                            imports_map.insert(import_source, Some(new_set));
-                        }
-                        _ => {
-                            if specs_name.is_empty() {
-                                imports_map.insert(import_source, None);
-                            } else {
-                                let new_set: HashSet<String> =
-                                    HashSet::from_iter(specs_name.iter().cloned());
-                                imports_map.insert(import_source, Some(new_set));
-                            }
-                        }
-                    }
+    let mut final_imports_map: HashMap<String, Option<HashSet<String>>> = HashMap::new();
+    visitor
+        .imports_map
+        .iter()
+        .for_each(|(k, v)| match final_imports_map.get_mut(k) {
+            Some(Some(specifiers)) => {
+                for spec in v {
+                    specifiers.insert(spec.clone());
+                }
+            }
+            Some(None) | None => {
+                if !v.is_empty() {
+                    final_imports_map.insert(k.clone(), Some(v.clone()));
                 }
             }
+        });
+    visitor.import_paths.iter().for_each(|path| {
+        if !final_imports_map.contains_key(path) {
+            final_imports_map.insert(path.clone(), None);
         }
     });
-    imports_map
-}
-
-fn get_specifier_name(fm: &SourceFile, spec: &swc_ecma_ast::ImportSpecifier) -> Option<String> {
-    if let Some(default) = spec.as_default() {
-        return Some(get_string_of_span(
-            &fm.src.as_bytes().to_vec(),
-            &default.span,
-        ));
-    }
-    if let Some(named) = spec.as_named() {
-        return Some(get_string_of_span(&fm.src.as_bytes().to_vec(), &named.span));
-    }
-    None
-}
-
-fn get_string_of_span<'a>(file_text: &'a Vec<u8>, span: &'a swc_common::Span) -> String {
-    String::from_utf8_lossy(&file_text[span.lo().to_usize() - 1..span.hi().to_usize() - 1])
-        .to_string()
+    visitor.require_paths.iter().for_each(|path| {
+        if !final_imports_map.contains_key(path) {
+            final_imports_map.insert(path.clone(), None);
```

Comment:
I think this has a bug. Analyzing this:
```ts
import { name } from 'foo'
require('foo')
```

Will produce:
```rust
map![
   "foo" => None,
]
```
instead of 
```rust
map![
   "foo" => Some(vec!["name"]),
]
```